### PR TITLE
Guide user setup connection string from UI

### PR DESCRIPTION
**Pull request by SLdragon**

### CODE REVIEW

File: src/common/constants.ts [hunk 1]
```diff
@@ -165,9 +165,12 @@ export class Constants {
     public static installFailedMsg = "'iotedgehubdev' tool installation has failed. Please install it manually for IoT Edge Simulator.";
     public static installManuallyMsg = "Please install 'iotedgehubdev' tool first for IoT Edge Simulator.";
     public static queryASAJobInfoFailedMsg = "The maximum retry count has been exceeded with empty response from the Stream Analytics.";
+    public static needSetupSimulatorMsg = "Please setup iotedgehubdev first before try to start simulator.";
```
Comment (adashen) [inline src/common/constants.ts]: before starting simulator

File: src/edge/simulator.ts [hunk 2]
```diff
@@ -223,13 +225,44 @@ export class Simulator {
         });
     }
 
+    private async checkAndGuideUserToSetupIotedgehubdev(outputChannel: vscode.OutputChannel) {
+        if (await this.isValidateConfigSupported()) {
+            try {
+                await Executor.executeCMD(null, "iotedgehubdev", { shell: true }, "validateconfig");
+            } catch (error) {
```
Comment (adashen) [inline src/edge/simulator.ts]: will we check error code?

Comment (adashen) [reply src/edge/simulator.ts]: If we have error code, do we need is versionsupported ?

Comment (SLdragon) [reply src/edge/simulator.ts]: The error code is always -1 for current version

File: src/edge/simulator.ts [hunk 3]
```diff
@@ -175,6 +175,7 @@ export class Simulator {
 
     public async startEdgeHubSingleModule(outputChannel: vscode.OutputChannel): Promise<void> {
         return await this.callWithInstallationCheck(outputChannel, async () => {
+            await this.checkAndGuideUserToSetupIotedgehubdev(outputChannel);
```
Comment (adashen) [inline src/edge/simulator.ts]: from the current flow, it will setup and continue to start?

File: src/common/constants.ts (hunk 1 above)
Comment (SLdragon) [reply src/common/constants.ts]: Thanks, updated

File: src/edge/simulator.ts (hunk 3 above)
Comment (SLdragon) [reply src/edge/simulator.ts]: updated

File: src/extension.ts [hunk 4]
```diff
@@ -246,6 +246,8 @@ function initCommandAsync(context: vscode.ExtensionContext,
         let errorData: ErrorData | undefined;
         const properties: { [key: string]: string; } = {};
         properties.result = "Succeeded";
+        properties.fromCommandPalette = (!args || !args[0]).toString();
```
Comment (adashen) [inline src/extension.ts]: Can we make sure all the context menu triggered has args[0] not null?

Comment (SLdragon) [reply src/extension.ts]: Yes, args[0] for all commands not triggered from command palette is not null

**Pull request merged**